frontend/src/components/CustomTable.tsx

```diff
@@ -206,6 +203,36 @@ interface CustomTableState {
   tokenList: string[];
 }
 
+interface CustomTableRowProps {
+  row: Row;
+  columns: Column[];
+}
+
+function calculateColumnWidths(columns: Column[]): number[] {
+  const totalFlex = columns.reduce((total, c) => total += (c.flex || 1), 0);
+  return columns.map(c => (c.flex || 1) / totalFlex * 100);
+}
+
+export const CustomTableRow: React.FC<CustomTableRowProps> = (props: CustomTableRowProps) => {
+  const {row, columns} = props;
+  const widths = calculateColumnWidths(columns);
+  return (
+    <React.Fragment>
+      {
+        row.otherFields.map((cell, i) => (
+          <div key={i} style={{width: widths[i] + '%'}} className={css.cell}>
+            {i === 0 && row.error && (
+              <Tooltip title={row.error}><WarningIcon className={css.icon} /></Tooltip>
+            )}
+            {columns[i].customRenderer ?
+              columns[i].customRenderer!({value: cell, id: row.id}) : cell}
```

Comment:
is the `!` still needed with the ternary operator?

Reply:
Surprisingly, it does. It seems like the compiler should be able to determine that it's being checked, but it doesn't seem to work.